File: Rotate Array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Rule: Do not return anything, modify nums in-place instead.
        """
        k = k % len(nums)
        nums[:] = nums[::-1]
        nums[:] = nums[0:k][::-1] + nums[k:][::-1]

File: javascript solutions/rotate array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    
    if (nums.length > k) {
        let len = nums.length - k;
        let split = nums.splice(len);
        nums.unshift(...split);
    }
    else {
        let i = 0;
        while (i < k) {
            nums.unshift(nums.pop());
            i++;
        }
    }
};
